### ISSUE TITLE
Output 'ItemList' structured data for archive pages

### ISSUE THREAD

**Issue by zitrusblau** [sybrew/The-SEO-Framework-Extension-Manager#14]
What about adding more article related structured data on archive views (categories, tags, custom taxonomies and custom post type archives)?

That would create rich search results like carousels for google (https://developers.google.com/search/docs/data-types/carousel)

Also, currently, on a custom post type archive, TSF outputs structured data of the first post in the list.

**Comment by sybrew** [comment]
I don't think we can reliably assert a post type's purpose, and create correct, non-penalizable data from that. As such, I don't believe this is feasible, and the post type creator would better handle such output.

This philosophy is already followed by us when it comes to WooCommerce, where we allow them to create structured data for Products. Sometimes, we file reports to them on correctness. After all, they handle the whole product framework, and they can maintain that much better.

At the very least, we could create Blocks which allow you to fill in such data. This will require both back and front-end development with active maintenance. That, of course, takes the time of which we, for now, have little to spare.

What I'm trying to say is that we can do everything imaginable with code, but we must know our limitations and focus on what we can handle best.

In any case, thanks for your bug report! I've marked it internally for investigation.

**Comment by zitrusblau** [comment]
Thanks for this. I understand that it's not a high priority or even intended to support this in TSF. And focusing on what really matters and don't bloat the codebase is always higher in priority from a developers point of view (that certainly also needs to maintain the code).

But given that you are about to release a more full fledged plugin API, this might be something a 'framework' for SEO might be able to provide.

At least, the JSON LD output should be corrected on archive pages, yes.

Anyways, if you ever happen to add basic support for this on which can be built upon, let me know.

**Comment by sybrew** [comment]
> Also, currently, on a custom post type archive, TSF outputs structured data of the first post in the list.

I was unable to reproduce this error on 3.2.4. Please hold out a little longer for the next update, where our more strict query checking may resolve this issue, regardless. If it doesn't, shoot me a private support ticket including the affected page URL. Thanks!

**Comment by sybrew** [comment]
> Also, currently, on a custom post type archive, TSF outputs structured data of the first post in the list.

I found this to be the case on Author archives, where, indeed the oldest post is used to make up this data. I'll make the checks more strict for the next update.
